### PR TITLE
[ODS-4391] Convert EdFi.Common.UnitTests to .NET Core

### PR DESCRIPTION
**Pull request by raghuramakrishna**

### CODE REVIEW

File: tests/EdFi.Common.UnitTests/Configuration/ConfigurationReaderTests.cs [hunk 1]
```diff
@@ -17,9 +19,16 @@ public class When_configuration_value_is_not_present
             [Test]
             public void Should_return_a_null_value()
             {
-                new AppConfigValueProvider().GetValue("Non-existent Key")
+                var config = new ConfigurationBuilder()
+                   .SetBasePath(TestContext.CurrentContext.TestDirectory)
+                   .AddJsonFile("appsettings.json", optional: true)
+                   .AddEnvironmentVariables()
+                   .Build();
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Configuration/ConfigurationReaderTests.cs:26]:
```suggestion
                var config = new ConfigurationBuilder()
                   .SetBasePath(TestContext.CurrentContext.TestDirectory)
                   .AddJsonFile("appsettings.json", optional: true)
                   .Build();
```

File: tests/EdFi.Common.UnitTests/Configuration/AppConfigValueProviderTests.cs [hunk 2]
```diff
@@ -27,7 +29,13 @@ public abstract class Fixture
             [SetUp]
             public void SetUp()
             {
-                SystemUnderTest = new AppConfigValueProvider();
+                var config = new ConfigurationBuilder()
+                   .SetBasePath(TestContext.CurrentContext.TestDirectory)
+                   .AddJsonFile("appsettings.json", optional: true)
+                   .AddEnvironmentVariables()
+                   .Build();
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Configuration/AppConfigValueProviderTests.cs:36]:
```suggestion
                var config = new ConfigurationBuilder()
                   .SetBasePath(TestContext.CurrentContext.TestDirectory)
                   .AddJsonFile("appsettings.json", optional: true)
                   .Build();
```

File: tests/EdFi.Common.UnitTests/Extensions/IEnumerableExtensionTests.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Extensions/IEnumerableExtensionTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Extensions/StringExtensionTests.cs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Extensions/StringExtensionTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Inflection/InflectorTests.cs [hunk 5]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Inflection/InflectorTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Security/PackedHashConverterTests.cs [hunk 6]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Security/PackedHashConverterTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Security/Pbkdf2HmacSha1SecureHasherTests.cs [hunk 7]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Security/Pbkdf2HmacSha1SecureHasherTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Security/SecurePackedHashProviderTests.cs [hunk 8]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Security/SecurePackedHashProviderTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Utils/Extensions/StringExtensionTests.cs [hunk 9]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Utils/Extensions/StringExtensionTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/Utils/Resources/EmbeddedResourceReaderTests.cs [hunk 10]
```diff
@@ -3,6 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+#if NETCOREAPP
```
Comment (jamessmckay) [inline tests/EdFi.Common.UnitTests/Utils/Resources/EmbeddedResourceReaderTests.cs:6]: why is this wrapped with netcoreapp?

File: tests/EdFi.Common.UnitTests/EdFi.Common.UnitTests.csproj [hunk 11]
```diff
@@ -11,27 +11,38 @@
...
+    <None Remove="appsettings.json" />
+    <Content Include="appsettings.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+    </Content>
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' ">
     <PackageReference Include="FakeItEasy" Version="6.2.0" />
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
[the remark this answers is not in the document]
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/EdFi.Common.UnitTests.csproj]: Done

File: tests/EdFi.Common.UnitTests/Configuration/ConfigurationReaderTests.cs (hunk 1 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Configuration/ConfigurationReaderTests.cs]: Done

File: tests/EdFi.Common.UnitTests/Configuration/AppConfigValueProviderTests.cs (hunk 2 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Configuration/AppConfigValueProviderTests.cs]: Done

File: tests/EdFi.Common.UnitTests/Extensions/IEnumerableExtensionTests.cs (hunk 3 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Extensions/IEnumerableExtensionTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Extensions/StringExtensionTests.cs (hunk 4 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Extensions/StringExtensionTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Security/PackedHashConverterTests.cs (hunk 6 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Security/PackedHashConverterTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Security/Pbkdf2HmacSha1SecureHasherTests.cs (hunk 7 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Security/Pbkdf2HmacSha1SecureHasherTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Security/SecurePackedHashProviderTests.cs (hunk 8 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Security/SecurePackedHashProviderTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Utils/Extensions/StringExtensionTests.cs (hunk 9 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Utils/Extensions/StringExtensionTests.cs]: removed

File: tests/EdFi.Common.UnitTests/Utils/Resources/EmbeddedResourceReaderTests.cs (hunk 10 above)
Comment (raghuramakrishna) [reply tests/EdFi.Common.UnitTests/Utils/Resources/EmbeddedResourceReaderTests.cs]: removed

**Pull request merged**